Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1148,38 +1141,38 @@ void GCMemcard::Gcs_SavConvert(DEntry& tempDEntry, int saveType, u64 length)
     // 0x34 and 0x35, 0x36 and 0x37, 0x38 and 0x39, 0x3A and 0x3B,
     // 0x3C and 0x3D,0x3E and 0x3F.
     // It seems that sav files also swap the banner/icon flags...
-    ByteSwap(&tempDEntry.m_unused_1, &tempDEntry.m_banner_and_icon_flags);
+    std::swap(tempDEntry.m_unused_1, tempDEntry.m_banner_and_icon_flags);
 
     std::array<u8, 4> tmp;
-    memcpy(tmp.data(), &tempDEntry.m_image_offset, 4);
-    ByteSwap(&tmp[0], &tmp[1]);
-    ByteSwap(&tmp[2], &tmp[3]);
-    memcpy(&tempDEntry.m_image_offset, tmp.data(), 4);
+    std::memcpy(tmp.data(), &tempDEntry.m_image_offset, 4);
+    std::swap(tmp[0], tmp[1]);
```

Comment:
This isn't a big endian to little endian swap for sure, so what is it?

Reply:
I'm not 100% sure as this isn't spelled out, but I suspect this happens because some thirdparty Memory Card dumping hardware (Datel's MAX Drive, I think?) reads these fields in 16 bits words of the wrong endianness. But either way, that's just how the SAV format stores its data.

Reply:
Fair enough, just wanted to point it out in an unlikely chance of a typo/misworded intention.

Reply:
Is this worth updating the comment above to include a (speculative?) reason _why_ this is? It does say "swap byte pairs", but gives no reason for it.